lib/routing/sfRoute.class.php

```diff
@@ -858,4 +858,35 @@ public function unserialize($data)
     list($this->tokens, $this->defaultOptions, $this->options, $this->pattern, $this->staticPrefix, $this->regex, $this->variables, $this->defaults, $this->requirements, $this->suffix, $this->customToken) = unserialize($data);
     $this->compiled = true;
   }
+
+    public function jsonSerialize(): array
+    {
+        return [
+            'tokens'         => $this->tokens,
+            'defaultOptions' => $this->defaultOptions,
+            'options'        => $this->options,
+            'pattern'        => $this->pattern,
+            'staticPrefix'   => $this->staticPrefix,
+            'regex'          => $this->regex,
+            'variables'      => $this->variables,
+            'defaults'       => $this->defaults,
+            'requirements'   => $this->requirements,
+            'suffix'         => $this->suffix,
+            'customToken'    => $this->customToken,
+        ];
+    }
+
+    public static function jsonUnserialize(array $raw): self
+    {
+        $rebuilt = new self(
+            $raw['pattern'],
+            $raw['defaults'],
+            $raw['requirements'],
+            $raw['options']
+        );
+
+        $rebuilt->compile();
+
+        return $rebuilt;
+    }
```

Comment:
So as discussed above, this method **won't** be automatically called when deserialising the cached content. I've named the method `jsonUnserialize()`, so that it matches the pattern of:
- `serialize()` -> `jsonSerialize()`
- `unserialize()` -> `jsonUnserialize()`

(even though it doesn't really do anything with JSON or deserialization). Strictly speaking, because this method has to be called manually, we can name it anything we want, so if you feel this ought to be renamed then please let me know.